client/src/module/authentication/container/register-container.spec.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from "react";
+import * as Enzyme from "enzyme";
+import { BrowserRouter as Router } from "react-router-dom";
+import RegisterContainer from "../container/register-container";
+import * as fetchUtils from "../../../util/fetch-util";
+import * as fetchMock from "fetch-mock";
+
+const validData = {
+	email: "test@test.com",
+	name: "tester",
+	password: "passPASSpass",
+	confirmPassword: "passPASSpass",
+};
+
+function fillOutForm(wrapper) {
+	const nameField = wrapper.find('input[name="name"]');
+	const emailField = wrapper.find('input[name="email"]');
+	const passwordField = wrapper.find('input[name="password"]');
+	const passwordConfirmField = wrapper.find('input[name="confirmPassword"]');
+
+	expect(nameField).toHaveLength(1);
+	expect(emailField).toHaveLength(1);
+	expect(passwordField).toHaveLength(1);
+	expect(passwordConfirmField).toHaveLength(1);
+
+	nameField.instance().value = validData.name;
+	nameField.simulate("change").simulate("blur");
+
+	emailField.instance().value = validData.email;
+	emailField.simulate("change").simulate("blur");
+
+	passwordField.instance().value = validData.password;
+	passwordField.simulate("change").simulate("blur");
+
+	passwordConfirmField.instance().value = validData.confirmPassword;
+	passwordConfirmField.simulate("change").simulate("blur");
+}
+
+function submitForm(wrapper) {
+	// Finds the form on the container and simulate hitting the 'register' button
+	const form = wrapper.find("form");
+	expect(form).toHaveLength(1);
+	form.simulate("submit");
+	// Wait a few seconds to the events to propagate - there are no promises we can wait for
+	// at the wrapper level to see if this is done so this is the next best thing for now
+	return new Promise((resolve) => setTimeout(resolve, 4000));
```

Comment:
It would be nice to find a way around this since the pattern will make things slower and slower the more tests that use it.  

One idea would be to move the submit function declaration out of the render method and make it a method on RegisterContainer.  You could then spy on RegisterContainer.submit in tests that are checking that the submit event does the right thing.  

You'd need to write separate tests to check that the submit method does the right thing but that should be easier because you wouldn't have to worry about enzyme getting in the way.